script.js

```diff
@@ -1,66 +1,151 @@
-    // this function creates just the task object with the relevent event handlers, without inner elements
-    let newTask = document.createElement("div");
-    newTask.classList.add("task")
-    return newTask;
-}
+    }
+
+    getState(){
+        return {
+            text: this.taskInputpreviousText,
+            done: this.isDone(),
+        };
+    }
 
-function addTaskInputBar(task) {
-    // this function will add to the task the input bar section, and will return the input bar object.
-    let taskInput = document.createElement("input");
-    taskInput.classList.add("task-input-text");
-    taskInput.addEventListener("keydown", (event) => {
-        if (event.key === "Enter") {
-            taskInput.blur();
+    setState(state){
+        this.taskInputpreviousText = state.text;
+        this.taskInput.value = this.taskInputpreviousText;
+        if (state.done){
+            this.markDone();
         }
-    })
-    task.appendChild(taskInput);
-    return taskInput;
+    }
+
+    handleKeydownEvent(event){
+        this.text = this.taskInput.value;
+        switch(event.key){
+            case "Enter":
+                this.taskInputpreviousText = this.taskInput.value;
+                this.taskInput.blur();
+                break;
+
+            case "Escape":
+                this.taskInput.blur();         
+        }
+    }
+
+    addTaskInputBar() {
+        this.taskInput = document.createElement("input");
+        this.taskInput.classList.add("task-input-text");
+        this.taskInput.addEventListener("keydown", (event) => {this.handleKeydownEvent(event)})
+        this.taskInput.addEventListener("blur", (event) => {this.taskInput.value = this.taskInputpreviousText})
+        this.div.appendChild(this.taskInput);
+    }
+
+    addTaskDeleteButton() {
+        this.deleteButton = document.createElement("button");
+        this.deleteButton.textContent = "X";
+        this.deleteButton.classList.add("delete-button");
+        this.deleteButton.onclick = () => {this.deleteTask();};
+        this.div.appendChild(this.deleteButton);
+    }
+
+    addTaskFinishedButton() {
+        this.finishedButton = document.createElement("button");
+        this.finishedButton.textContent = '\u{02713}';
+        this.finishedButton.classList.add("finished-button");
+        this.finishedButton.onclick = () => { this.changeTaskDoneMark(this.taskInput) };
+        this.div.appendChild(this.finishedButton);
+    }
+
+    markDone(){
+        if(!this.isDone()){
+            this.taskInput.classList.add(DONE_TASK_TEXT_CLASS);
+        }
+    }
+
+    unMarkDone(){
+        if(this.isDone()){
+            this.taskInput.classList.remove(DONE_TASK_TEXT_CLASS);
+        }
+    }
+
+    changeTaskDoneMark() {
+        if (this.isDone()) {
+            this.unMarkDone()
+        } else {
+            this.markDone()
+            moveToBottom(this);
+        }
+    }
+
+    isDone(){
+        return this.taskInput.classList.contains(DONE_TASK_TEXT_CLASS);
```

Comment:
The class describes how the element is displayed, doesn`t hold information about the element.

Reply:
done